agent/config/config.go

```diff
@@ -311,7 +311,18 @@ func environmentConfig() (Config, error) {
 	taskENIEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_ENI"), false)
 	taskIAMRoleEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE"), false)
 	taskIAMRoleEnabledForNetworkHost := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST"), false)
-	taskCPUMemLimitEnabled := utils.ParseBool(os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT"), true)
+
+	var taskCPUMemLimitEnabled Conditional
+	taskCPUMemLimitConfigString := os.Getenv("ECS_ENABLE_TASK_CPU_MEM_LIMIT")
+
+	switch {
+	case taskCPUMemLimitConfigString == "":
+		taskCPUMemLimitEnabled = DefaultEnabled
```

Comment:
This logic shouldn't be here.  If you see an explicit enable or explicit disable, use that.  Defaults should be in the existing defaults method though, and you should go through the merge for that.

Reply:
Ah, good point!